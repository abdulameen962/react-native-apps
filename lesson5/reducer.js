import {combineReducers} from 'redux'
import {UPDATE_CONTACT,UPDATE_USER,LOG_IN_FULFILLED,LOG_IN_SENT,LOG_IN_REJECTED} from './actions'

const merge = (prev, next) => Object.assign({}, prev, next)

const contactReducer = (state = [], action) => {
  if (action.type === UPDATE_CONTACT) return [...state, action.payload]
  return state
}

const userReducer = (state = {}, action) => {

  switch (action.type){
    case UPDATE_USER:
        return merge(state, action.payload)

    case UPDATE_CONTACT:
        return merge(state, {prevContact: action.payload})

    case LOG_IN_FULFILLED:
        return merge(state,{token:action.payload})

    case LOG_IN_REJECTED:
        return merge(state,{loginErr:action.payload})

    default:
        return state
  } 
}

// const reducer = (state, action) => ({
//   user: userReducer(state.user, action),
//   contacts: contactReducer(state.contacts, action),
// })

const reducer = combineReducers({
    user: userReducer,
    contact: contactReducer
})

export default reducer