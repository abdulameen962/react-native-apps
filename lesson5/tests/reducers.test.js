import reducer from '../reducer'

import * as actions from '../actions'

const DEFAULT_STATE = {
    user:{},
    contact:[],
}

describe('contact reducer', () => {
    it('sucessfully adds new contact',() => {
        expect(reducer(DEFAULT_STATE,actions.addContact({name:'test user',phone:'1234567'}))).toMatchSnapshot();

    })
})

describe('user reducer', () => {
    it('sucessfully adds new user',() => {
        expect(reducer(DEFAULT_STATE,actions.updateUser({name:'test user'}))).toMatchSnapshot();

    })
})
