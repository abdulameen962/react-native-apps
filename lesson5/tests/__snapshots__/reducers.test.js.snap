// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact reducer sucessfully adds new contact 1`] = `
{
  "contact": [
    {
      "name": "test user",
      "phone": "1234567",
    },
  ],
  "user": {
    "prevContact": {
      "name": "test user",
      "phone": "1234567",
    },
  },
}
`;

exports[`user reducer sucessfully adds new user 1`] = `
{
  "contact": [],
  "user": {
    "name": "test user",
  },
}
`;
